config/config_test.go

```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/stretchr/testify/require"
```

Comment:
These unit tests are good! But we will also want some [integration tests](https://github.com/gruntwork-io/terragrunt/tree/master/test) though. At a minimum, I'd like to see tests for what happens when:

- Terragrunt module only has `terragrunt.hcl.json` and you run `terragrunt plan` (no args to specify the different file).
- Multiple terragrunt modules only have `terragrunt.hcl.json` and you run `terragrunt plan-all`
- Multiple terragrunt modules with a mix of `terragrunt.hcl` and `terragrrunt.hcl.json` and you run `terragrunt plan-all`

Some more interesting test cases for to be comprehensive:

- Child terragrunt module that includes a parent terragrunt config that is in JSON format, and vice versa.
- Terragrunt module that marks a json based terragrunt module as a dependency in a `dependency` block, and vice versa.
- Terragrunt module that marks a json based terragrunt module as a dependency in a `dependencies` block, and vice versa. Also when there is a mix of json and hcl based modules as `dependency` and `dependencies` blocks.

Reply:
@yorinasub17 For the first integration test, TerragruntOptions default the value of terragruntConfigPath to be `terragrunt.hcl` at https://github.com/gruntwork-io/terragrunt/blob/master/cli/args.go#L66.

A couple of questions:
- What is the desired behavior? Error out?
- Should `parseTerragruntOptionsFromArgs` interrogate the files in the working path to determine whether we set the default config to be .hcl or .hcl.json ?
  - If so, which file should take precedence? Or should it error if both are found?

```
[terragrunt] 2019/11/11 10:58:13 Reading Terragrunt config file at fixture-config-files/one-json-config/subdir/terragrunt.hcl
--- FAIL: TestTerragruntWorksWithSingleJsonConfig (0.00s)
    c:\go_home\go\src\github.com\gruntwork-io\terragrunt\test\integration_test.go:2248: Failed to run Terragrunt command 'terragrunt plan --terragrunt-non-interactive --terragrunt-working-dir fixture-config-files/one-json-config/subdir' due to error: Error reading file at path fixture-config-files/one-json-config/subdir/terragrunt.hcl: open fixture-config-files/one-json-config/subdir/terragrunt.hcl: The system cannot find the file specified.
```

Reply:
I think the logic should be to look for `terragrunt.hcl`, and if it doesn't exist, use `terragrunt.hcl.json`. I think `terragrunt.hcl` should take precedence for the same reasons mentioned in https://github.com/gruntwork-io/terragrunt/pull/863#discussion_r324063211.